test/it/client-delete-application-user.js

```diff
@@ -0,0 +1,75 @@
+const expect = require('chai').expect;
+
+const okta = require('../../');
+const utils = require('../utils');
+
+let orgUrl = process.env.OKTA_CLIENT_ORGURL;
+
+if (process.env.OKTA_USE_MOCK) {
+  orgUrl = `${orgUrl}/application-delete-user`;
+}
+
+const client = new okta.Client({
+  orgUrl: orgUrl,
+  token: process.env.OKTA_CLIENT_TOKEN
+});
+
+describe('client.deleteApplicationUser()', () => {
+
+  it('should allow me to remove a user from a application', async () => {
+    const application = {
+      name: 'bookmark',
+      label: 'my bookmark app',
+      signOnMode: 'BOOKMARK',
+      settings: {
+        app: {
+          requestIntegration: false,
+          url: 'https://example.com/bookmark.htm'
+        }
+      }
+    };
+
+    const user = {
+      profile: {
+        firstName: 'John',
+        lastName: 'Activate',
+        email: 'john-activate@example.com',
+        login: 'john-activate@example.com'
+      },
+      credentials: {
+        password: { value: 'Abcd1234' }
+      }
+    };
+
+    let createdApplication;
+    let createdUser;
+    let createdAppUser;
+
+    try {
+      await utils.removeAppByLabel(client, application.label);
+      await utils.cleanup(client, user);
+      createdApplication = await client.createApplication(application);
+      createdUser = await client.createUser(user);
+      createdAppUser = await client.assignUserToApplication(createdApplication.id, {
+        id: createdUser.id
+      });
+      await client.deleteApplicationUser(createdApplication.id, createdAppUser.id)
+      .then(result => {
+        expect(result.status).to.equal(204);
```

Comment:
I'm trying to remember, do we only return this status property on delete? I thought there was some nesting ie. `result.httpResponse.status`

Reply:
Yeah, right now we just return the http response for delete operations.  It's not consistent with other operations, and I want to change this (at which point these tests will also have to change)